modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseBuilder.java

```diff
@@ -167,6 +167,16 @@ else if (!configuredContentType.startsWith(HttpHeaders.Values.APPLICATION_X_WWW_
                     warnMapPayloadButNoUrlEncodedContentType(httpResponseHeaderBuilder.getContentType());
                 }
                 httpEntity = createUrlEncodedEntity(event, (Map) payload);
+
+                if (responseStreaming == ALWAYS && supportsTransferEncoding(event))
+                {
+                    setupChunkedEncoding(httpResponseHeaderBuilder);
+                }
+                else
+                {
+                    ByteArrayHttpEntity byteArrayHttpEntity = (ByteArrayHttpEntity) httpEntity;
```

Comment:
Yes, but that is in the case where a form request is sent to the listener, in which case it propagates a NullPayload through the flow which then makes it back to be sent as a response. I think you need to consider the scenario where the payload changes to an empty map mid flow so that reaches the response builder.